ecodev_core/filter_db_filters.py

```diff
@@ -0,0 +1,283 @@
+"""
+Low level methods to retrieve data from db in a paginated way, using a filter model
+"""
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ecodev_core import engine
+from ecodev_core import logger_get
+from ecodev_core import ServerSideField
+from ecodev_core import ServerSideFilter
+from ecodev_core.db_filters import SERVER_SIDE_FILTERS
+from sqlalchemy import func
+from sqlalchemy.orm import InstrumentedAttribute
+from sqlmodel import col
+from sqlmodel import or_
+from sqlmodel import select
+from sqlmodel import Session
+from sqlmodel.sql.expression import SelectOfScalar
+
+log = logger_get(__name__)
+
+
+def count_rows(
+        model: Any,
+        filter_model: Optional[Dict],
+        search_str: str = '',
+        search_cols: Optional[List] = None,
+) -> int:
+    """
+    Count the total number of rows, with reference to a filtering model
+    """
+    full_query = _get_full_query(
+        model=model,
+        filter_model=filter_model,
+        start_row=None,
+        end_row=None,
+        search_str=search_str,
+        search_cols=search_cols,
+        fields_order=None,
+        count=True
+    )
+
+    with Session(engine) as session:
+        return session.exec(full_query).one()
+
+
+def get_rows(
+        model: Any,
+        filter_model: Optional[Dict],
+        start_row: Optional[int] = None,
+        end_row: Optional[int] = None,
+        search_str: str = '',
+        search_cols: Optional[List] = None,
+        fields_order: Optional[Callable] = None,
+) -> List[Dict[str, Any]]:
+    """
+    Select relevant row lines from model db based on provided filter model.
+
+    Breaking change from previous version:
+    * filter_str is removed
+    * Keeps the column names from the database (separate front and back)
+    * does not use Pandas
+    * no filtering of columns (in particular: no argument 'fields')
+
+    NB:
+    * Select the whole db if no start_row or end_row is provided.
+    * 'fields_order' specify how to order the result rows
+    * 'search_str' corresponds to the search string from the search input.
+
+    """
+    full_query = _get_full_query(
+        model=model,
+        filter_model=filter_model,
+        start_row=start_row,
+        end_row=end_row,
+        search_str=search_str,
+        search_cols=search_cols,
+        fields_order=fields_order,
+        count=False
+    )
+
+    with Session(engine) as session:
+        data_list = [row.model_dump() for row in session.exec(full_query).all()]
+
+    return data_list
+
+
+def get_unique_values(
```

Comment:
What is it for? 

All the methods in this module should be thouroughly tested :). Look at the existing tests to (hopefully :)) get some help!